### ISSUE TITLE
LC0005 false positive

### ISSUE THREAD

**Issue by Aize-S** [StefanMaron/BusinessCentral.LinterCop#472]
In one of my reports, I get this LC0005 message:
![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/24434095/57902b46-942b-4ff1-8470-75f39b13246c)

Global var:
![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/24434095/de80940c-22c8-4295-8acd-f7dc86acc930)

The line of code in question:
![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/24434095/2e4aba22-f30d-4cc0-8620-24af8cc3fcf1)

Definition of the procedure that is called:
![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/24434095/8b05f47a-74b0-4135-b256-dbedc3e411e1)

When adhering to the warnings wishes:
![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/24434095/1841a2fc-3325-4704-b69c-c1b63082aa2f)

The error changes:
![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/24434095/69e20761-51bc-41ce-b460-79438cbd0a87)

I.e. I'm running in circles 😉

Changing the procedure definition solves my problem for now:
![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/24434095/0e6f2bd6-1e95-4041-be3b-9157f85b76d3)

**Comment by Arthurvdv** [comment]
I've created an repo and indeed the rule creates an false positive on this scenario.

![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/44637996/85b470a3-7a4b-428b-9dd9-f6fdf605604e)

The rule detects `LABRegistration` and `LabRegistration` as a valid IdentifierName and matches this with the declared identifies. Unfortunately at the moment I'm not sure how I could resolve this.

**Comment by Arthurvdv** [comment]
Repo example

```AL
table 50100 "LAB Registration"
{
    fields { field(1; MyField; Integer) { } }
}

codeunit 50100 FormatAddress
{
    procedure LabRegistration(var LABRegistration: Record "LAB Registration")
    begin
    end;
}

codeunit 50101 MyCodeunit
{
    procedure MyProcedure()
    var
        LABRegistration: Record "LAB Registration";
        FormatAddress: Codeunit FormatAddress;
    begin
        FormatAddress.LabRegistration(LABRegistration);
    end;
}
```

**Comment by Aize-S** [comment]
> I've created an repo and indeed the rule creates an false positive on this scenario.
> 
> The rule detects `LABRegistration` and `LabRegistration` as a valid IdentifierName and matches this with the declared identifies. Unfortunately at the moment I'm not sure how I could resolve this.

Thanks for looking into it. It's probably not a scenario that happens a lot. If you think of a way to fix it; great! If not; so be it.

**Comment by Arthurvdv** [comment]
Hi @Aize-S,

I'm cleaning up the old issues here and I've did some work on this on the successor of the BusinessCentral.LinterCop and I would invite you to migrate to ALCOps, where these false positives are already solved and we’re continuing development there (see also the note on the [README](https://github.com/StefanMaron/BusinessCentral.LinterCop/blob/master/README.md)).

A starting point can be the [Migrating from BusinessCentral.LinterCop](https://alcops.dev/docs/lintercop-migration/) on the alcops.dev website. Also on the documentation are the setup for VS Code and integration with pipelines for GitHub and/Azure DevOps.

I'm going to close the issue here, but if you have any questions regarding ALCops, it would be helpful to create a discussion/issue on one of the repos of https://github.com/ALCops
